src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -5883,26 +5883,79 @@ public void ToString_DerivedRecord_AbstractSealed()
             Assert.True(toString.IsImplicitlyDeclared);
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void ToString_DerivedRecord_BaseHasSealedToString(bool usePreview)
+        {
+            var src = @"
+var c = new C2();
+System.Console.Write(c.ToString());
+
+record C1
+{
+    public sealed override string ToString() => ""C1"";
+}
+record C2 : C1;
+";
+
+            var comp = CreateCompilation(src, parseOptions: usePreview ? TestOptions.RegularPreview : TestOptions.Regular9, options: TestOptions.DebugExe);
+            if (usePreview)
+            {
+                comp.VerifyEmitDiagnostics();
+                CompileAndVerify(comp, expectedOutput: "C1");
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (4,35): error CS8652: The feature 'sealed record ToString' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     public sealed override string ToString() => throw null;
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "ToString").WithArguments("sealed record ToString").WithLocation(7, 35)
+                    );
+            }
+        }
+
         [Fact]
-        public void ToString_DerivedRecord_BaseHasSealedToString()
+        public void ToString_DerivedRecord_BaseBaseHasSealedToString()
```

Comment:
>ToString_DerivedRecord_BaseBaseHasSealedToString [](start = 20, length = 48)

I think we should add more tests for:
- trying to override the sealed method explicitly and how next level derived records are going to behave.
- trying to shadow method, including in a way when it will not be accessible in derived types 
- inheritance behvior in presence of shadowing up the hierarhy

Reply:
> * trying to override the sealed method explicitly and how next level derived records are going to behave.

Done in ToString_DerivedRecord_BaseBaseHasSealedToString_And_BaseTriesToOverride

> * trying to shadow method, including in a way when it will not be accessible in derived types, including when shadowed method is not sealed.

Done in ToString_DerivedRecord_BaseBaseHasSealedToString_And_BaseShadowsToStringPrivate and ToString_DerivedRecord_BaseBaseHasSealedToString_And_BaseShadowsToStringNonSealed

> * inheritance behvior in presence of shadowing up the hierarhy

I think this is already covered in ToString_NewToString_SealedBaseToString